packages/material-ui/src/ClickAwayListener/ClickAwayListener.js

```diff
@@ -5,82 +5,89 @@ import ReactDOM from 'react-dom';
 import PropTypes from 'prop-types';
 import EventListener from 'react-event-listener';
 import ownerDocument from '../utils/ownerDocument';
+import { useForkRef } from '../utils/reactHelpers';
+
+function useMountedRef() {
+  const mountedRef = React.useRef(false);
+  React.useEffect(() => {
+    mountedRef.current = true;
+    return () => {
+      mountedRef.current = false;
+    };
+  }, []);
+
+  return mountedRef;
+}
 
 /**
  * Listen for click events that occur somewhere in the document, outside of the element itself.
  * For instance, if you need to hide a menu when people click anywhere else on your page.
  */
-class ClickAwayListener extends React.Component {
-  mounted = false;
-
-  moved = false;
-
-  componentDidMount() {
-    // Finds the first child when a component returns a fragment.
-    // https://github.com/facebook/react/blob/036ae3c6e2f056adffc31dfb78d1b6f0c63272f0/packages/react-dom/src/__tests__/ReactDOMFiber-test.js#L105
-    this.node = ReactDOM.findDOMNode(this);
-    this.mounted = true;
-  }
-
-  componentWillUnmount() {
-    this.mounted = false;
-  }
-
-  handleClickAway = event => {
+function ClickAwayListener(props) {
+  const { children, mouseEvent, touchEvent, onClickAway, ...other } = props;
+  const mountedRef = useMountedRef();
+  const movedRef = React.useRef(false);
+
+  const nodeRef = React.useRef();
+  // can be removed once we drop support for non ref forwarding class components
+  const handleOwnRef = React.useCallback(ref => {
+    nodeRef.current = ReactDOM.findDOMNode(ref);
+  }, []);
+  const handleRef = useForkRef(children.ref, handleOwnRef);
+
+  function handleClickAway(event) {
```

Comment:
Could we memoize each callback method? I fear it will be a regression without #12613. I should have added tests! 😞

Reply:
Is EventListener a memoized component? It's not a regression with #12613 though as far as I can tell. In that PR the rerender was fixed by not passing children to EventListener.

Reply:
Just checked, it is. Will do